File: backoffice/src/diningTable/domain/events/DiningTableDeleted.py
"""
 *
 * Libraries 
 *
"""

from src.shared.domain            import DomainEvent
from ..valueObjects.DiningTableId import DiningTableId

"""
 *
 * Class
 *
"""

class DiningTableDeleted( DomainEvent ):

    """
     *
     * Attributes 
     *
    """

    __id : DiningTableId

    """
     *
     * Methods 
     *
    """

    def __init__( 
        self, 
        id        : DiningTableId = None,
        eventId   : str           = None,
        timestamp : int           = None
    ) -> None:
        super().__init__( eventId, timestamp )
        self.__id = id
    
    def getId( self ) -> DiningTableId:
        return self.__id
    
    def getEventName( self ) -> str:
        return 'dining.table.deleted'
    
    def toPrimitives( self ) -> dict[str, str | int]:
        # Variables
        data : dict[str, str | int]
        # Code
        data = {
            'id' : self.__id.getValue()
        }
        return data

    def fromPrimitives( self, 
        eventId   : str, 
        timestamp : int, 
        data      : dict[str, str | int]
    ) -> object:
        return DiningTableDeleted( DiningTableId( data['id'] ), eventId, timestamp )

File: preparation/src/diningTable/domain/events/DiningTableDeleted.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.diningTable.domain {

    public class DiningTableDeleted : DomainEvent {

        /* 
         * 
         * Attributes
         *
        */

        private DiningTableId id;

        /* 
         * 
         * Methods
         *
        */

        public DiningTableDeleted() {
        }

        public DiningTableDeleted( DiningTableId id ) {
            this.id = id;
        }

        public DiningTableDeleted(
            DiningTableId id,
            string        eventId,
            int           timestamp
        ) : base( eventId, timestamp ) {
            this.id = id;
        }

        public DiningTableId getId() {
            return id;
        }

        public override DomainEvent fromPrimitives(
            string                     eventId, 
            int                        timestamp, 
            Dictionary<string, object> data
        ) {
            // Variables
            DiningTableDeleted domainEvent;
            // Code
            domainEvent = new DiningTableDeleted(
                new DiningTableId( data["id"].ToString() ),
                eventId,
                timestamp
            );
            return domainEvent;
        }

        public override string getEventName() {
            return "dining.table.deleted";
        }

        public override Dictionary<string, object> toPrimitives() {
            // Variables
            Dictionary<string, object> data;
            // Code
            data = new Dictionary<string, object>();
            data.Add( "id", id.getValue() );
            return data;
        }

    }

}